### PR TITLE
New channel event types for new-conversations and referrals

### PR DESCRIPTION
**Pull request by nicpottier**
Will update the telegram PR accordingly after this gets merged.

### CODE REVIEW

**Comment by nicpottier** [comment]
This now refactors the channel even log a bunch. Please re-review @rowanseymour

File: temba/sql/current_functions.sql [hunk 1]
```diff
@@ -127,9 +127,7 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql;
 
-----------------------------------------------------------------------
 -- Trigger procedure to update system labels on channel event changes
-----------------------------------------------------------------------
```
Comment (rowanseymour) [inline temba/sql/current_functions.sql]: Minor point but I think we use this pattern for all function/trigger comments

File: temba/sql/current_functions.sql [hunk 2]
```diff
@@ -139,9 +137,7 @@ BEGIN
       RETURN NULL;
     END IF;
 
-    IF NEW.is_active THEN
```
Comment (rowanseymour) [inline temba/sql/current_functions.sql]: What's the reason for the switch away from using `is_active` on channel events?

Comment (nicpottier) [reply temba/sql/current_functions.sql]: Consistency with Msg for one, it just doesn't seem super useful to have is_active on them either as they really aren't a user created thing.

File: temba/sql/current_functions.sql (hunk 1 above)
Comment (nicpottier) [reply temba/sql/current_functions.sql]: Problem is when viewing the log of `collect_sql` it looks jank having these.

**Comment by nicpottier** [comment]
@rowanseymour one more time, then want to merge this

File: temba/orgs/models.py [hunk 3]
```diff
@@ -1552,7 +1552,10 @@ def add_credits(self, bundle, token, user):
                 for card in existing_cards:
                     card.delete()
 
-                card = customer.cards.create(card=token)
+                try:
+                    card = customer.cards.create(card=token)
+                except stripe.CardError:
+                    raise ValidationError(_("Sorry, your card was declined, please contact your provider or try another card."))
```
Comment (nicpottier) [inline temba/orgs/models.py]: Sneaking this in, this fixes an issue we were seeing in that we were showing a generic error when credit card validation failed on the Stripe side.

File: temba/channels/migrations/0078_auto_20170831_1830.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('channels', '0077_auto_20170824_1555'),
+        ('msgs', '0076_install_triggers'),
+    ]
+
+    operations = [
+        migrations.RunPython(delete_inactive_channelevents),
```
Comment (ericnewcomer) [inline temba/channels/migrations/0078_auto_20170831_1830.py]: Migrate duration to extra?

File: templates/contacts/contact_history.haml [hunk 5]
```diff
@@ -42,9 +42,14 @@
...
-            Missed call
+            -trans "Missed call"
+          -elif event.event_type == 'new_conversation'
+            -trans "Conversation Started"
+          -elif event.event_type == 'referral'
+            -trans "Referred"
           -else
-            Phone call ({{event.duration|format_seconds}})
+            -trans "Phone call"
+            ({{event.duration|format_seconds}})
```
Comment (ericnewcomer) [inline templates/contacts/contact_history.haml]: is there a corresponding get_duration to pull this from extra?

File: temba/channels/migrations/0078_auto_20170831_1830.py (hunk 4 above)
Comment (nicpottier) [reply temba/channels/migrations/0078_auto_20170831_1830.py]: doh, yes!

**Pull request merged**